File: frontend/src/types/userRole.ts
// E:\Audit\AMS\frontend\src\types\userRole.ts

export type UserRole = {
  id: number;
  user_id: number;
  role_id: number;
  is_active: boolean;

  user_login_id?: string | null;
  user_full_name?: string | null;
  user_email?: string | null;

  role_name?: string | null;
  role_description?: string | null;

  created_by: string | null;
  updated_by: string | null;
  created_at: string;
  updated_at: string;
};

export type UserRolePayload = {
  user_id: number;
  role_id: number;
};

export type UserRoleMessageResponse = {
  message: string;
  data: UserRole | null;
};

File: backend/app/schemas/user_role.py
# E:\Audit\AMS\backend\app\schemas\user_role.py

from datetime import datetime
from pydantic import BaseModel, ConfigDict

class UserRoleCreate(BaseModel):
    user_id: int
    role_id: int


class UserRoleResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    role_id: int
    is_active: bool

    user_login_id: str | None = None
    user_full_name: str | None = None
    user_email: str | None = None

    role_name: str | None = None
    role_description: str | None = None

    created_by: str | None = None
    updated_by: str | None = None
    created_at: datetime
    updated_at: datetime


class UserRoleMessageResponse(BaseModel):
    message: str
    data: UserRoleResponse | None = None